File: dolmisani.games.graphrace/src/dolmisani/games/graphrace/Position.java
package dolmisani.games.graphrace;

public class Position {

	private int x;
	private int y;
	
	public Position(int x, int y) {
		
		this.x = x;
		this.y = y;
	}

	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}
	
	public void moveTo(Step s) {
		this.x += s.getDeltaX();
		this.y += s.getDeltaY();
	}
	
	public void backFrom(Step s) {
		this.x -= s.getDeltaX();
		this.y -= s.getDeltaY();
	}
	
	@Override
	public String toString() {
		return String.format("(%d , %d)", x, y);
	}

}

File: pwa/js/position.js
class Position {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }

    getX() {
        return this.x;
    }

    setX(x) {
        this.x = x;
    }

    getY() {
        return this.y;
    }

    setY(y) {
        this.y = y;
    }

    moveTo(stepOrDeltaX, deltaY) {
        if (stepOrDeltaX instanceof Step) {
            this.x += stepOrDeltaX.getDeltaX();
            this.y += stepOrDeltaX.getDeltaY();
        } else {
            this.x += stepOrDeltaX;
            this.y += deltaY;
        }
    }

    toString() {
        return `(${this.x}, ${this.y})`;
    }
}
